enumerable.rb

```diff
@@ -0,0 +1,158 @@
+# rubocop:disable Layout/LineLength
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable  Metrics/MethodLength
+# rubocop:disable  Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    n = length
+    arr = *self
+    n.times do |i|
+      arr[i] = yield(arr[i])
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    n = length
+    n.times do |index|
+      yield(self[index], index)
+    end
+    self
+  end
```

Comment:
- [ ] when a block is given when self is a range it should return the range itself after calling the given block once for each element in self

EXAMPLE: range.my_each_with_index(&block)  `evens = (1..10).each_with_object([]) { |i, a| a << i*2 }
#=> [2, 4, 6, 8, 10, 12, 14, 16, 18, 20]` use this [link](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index)

- [ ] When you are given a Hash it should call the block once for each element in the hash. 

Example: `hash = Hash.new
%w(cat dog wombat).each_with_index { |item, index|
  hash[item] = index
}
hash   #=> {"cat"=>0, "dog"=>1, "wombat"=>2}`

- [ ]

Reply:
Dear Peter, we appreciate the very helpful review you gave us and we addressed them. But with all due respect, the .each_with_object method is not a requirement for this project.

Reply:
Hello @henatan99 sorry for the confusion. It is actually an error on my side. I actually meant each_with_index. Sorry again for the confusion and thank you for the polite correction.